src/plan/pageprotect/global.rs

```diff
@@ -168,6 +168,17 @@ impl<VM: VMBinding> PageProtect<VM> {
                 &CONSTRAINTS,
                 global_metadata_specs,
             ),
+        };
+
+        {
+            use crate::util::metadata::side_metadata::SideMetadataSanity;
+            let mut side_metadata_sanity_checker = SideMetadataSanity::new();
+            ret.common
+                .verify_side_metadata_sanity(&mut side_metadata_sanity_checker);
```

Comment:
Is this part of the extreme assertions or is it a cheap check?

Reply:
It is only called once per plan at boot time. I guess it is fine. `extreme_assertions` needs this check, but this check itself is not part of `extreme_assertions`.

Reply:
Maybe we should add some comments so that when people write new plans, they know to call `SideMetadataSanity`

Reply:
Yeah. I will do that.

Reply:
Sounds good. I have added some comments. But as `new()` is specific for each plan, I can only put some comments on the `Plan` trait type about a few things developers need to do in a plan constructor. I could not think of a better place to put those comments. If you think there is a better place to put those comments, please let me know.

Reply:
Potentially also the tutorial?

Reply:
The tutorial actually includes the constructor code that shows the side metadata sanity checker. I have added more comments to all the plan constructors (including the tutorial plan).